pkg/api/v1/conversion.go

```diff
@@ -623,10 +627,9 @@ func Convert_v1_Pod_To_api_Pod(in *Pod, out *api.Pod, s conversion.Scope) error
 		// back to the caller.
 		in.Spec.InitContainers = values
 		// Call defaulters explicitly until annotations are removed
-		for i := range in.Spec.InitContainers {
-			c := &in.Spec.InitContainers[i]
-			SetDefaults_Container(c)
```

Comment:
`SetDefaults_Container` is only set `ImagePullPolicy`; there's still other fields need to update according to https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/zz_generated.defaults.go#L216